cmake/modules/configuration_files.cmake

```diff
@@ -78,9 +78,15 @@ else()
   message(FATAL_ERROR "No prj.conf file was found in the ${APPLICATION_CONFIG_DIR} folder, "
                       "please read the Zephyr documentation on application development.")
 endif()
+zephyr_file_suffix(CONF_FILE)
 
 if(CONF_FILE_INCLUDE_FRAGMENTS)
-  zephyr_file(CONF_FILES ${APPLICATION_CONFIG_DIR}/boards KCONF CONF_FILE BUILD ${CONF_FILE_BUILD_TYPE})
+  # FILE_SUFFIX will take priority over a prj_<X> build type if both are used
```

Comment:
I think a better approach would be to disable the `prj_<X>` pattern up here:
https://github.com/zephyrproject-rtos/zephyr/blob/9102821fbe282d1b994ee21c0e3015c885de3d58/cmake/modules/configuration_files.cmake#L61

`if(NOT DEFINED FILE_SUFFIX AND ...`. I'm assuming we should deprecate that entire logic anyway, unless you see a reason to still support `prj_<X> -> prj_<X>_<suffix>`.